bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,135 @@
+                ±HH:MM
+                ±HHMM
+                ±HH
+
+        Times in the dropdown are shown in UTC
+        """
+        if not date_time:
+            await invoke_help_command(ctx)
+            return
+
+        if isinstance(date_time, tuple):
+            # Remove empty strings. Strip extra whitespace from the remaining items
+            ignored_tokens = list(map(str.strip, filter(str.strip, date_time[1])))
+            date_time = date_time[0]
+            if ignored_tokens:
+                await ctx.send(f"Could not parse the following token(s): `{', '.join(ignored_tokens)}`")
+            await ctx.send(f"Date and time parsed as: `{date_time.format(arrow.FORMAT_RSS)}`")
+
+        epoch = int(date_time.timestamp())
+        view = TimestampMenuView(ctx, self._format_dates(date_time), epoch)
+        original = await ctx.send(f"`{epoch}`", view=view)
+        await view.wait()  # wait until expiration before removing the dropdown
+        await original.edit(view=None)
+
+    @staticmethod
+    def _format_dates(date: arrow.Arrow) -> list[str]:
+        """
+        Return a list of date strings formatted according to the discord timestamp styles.
+
+        These are used in the description of each style in the dropdown
+        """
+        date = date.to('utc')
+        formatted = [str(int(date.timestamp()))]
+        formatted += [date.format(format[1]) for format in list(STYLES.values())[1:7]]
+        formatted.append(date.humanize())
+        return formatted
+
+
+class TimestampMenuView(discord.ui.View):
+    """View for the epoch command which contains a single `discord.ui.Select` dropdown component."""
+
+    def __init__(self, ctx: commands.Context, formatted_times: list[str], epoch: int):
+        super().__init__(timeout=DROPDOWN_TIMEOUT)
+        self.ctx = ctx
+        self.epoch = epoch
+        self.dropdown: discord.ui.Select = self.children[0]
+        for label, date_time in zip(STYLES.keys(), formatted_times):
+            self.dropdown.add_option(label=label, description=date_time)
+
+    @discord.ui.select(placeholder="Select the format of your timestamp")
+    async def select_format(self, _: discord.ui.Select, interaction: discord.Interaction) -> discord.Message:
+        """Drop down menu which contains a list of formats which discord timestamps can take."""
+        selected = interaction.data["values"][0]
+        if selected == "Epoch":
+            return await interaction.response.edit_message(content=f"`{self.epoch}`")
+        return await interaction.response.edit_message(content=f"`<t:{self.epoch}:{STYLES[selected][0]}>`")
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        """Check to ensure that the interacting user is the user who invoked the command."""
+        if interaction.user != self.ctx.author:
+            embed = discord.Embed(description="Sorry, but this interaction can only be used by the original author.")
```

Comment:
Similar comment I left on one of D0rs4n's PR, but I'm concerned about someone not familiar with d.py knowing what the term "interaction" necessarily means. Maybe something like "Sorry, this response can only be used by ...") but I'm also cool if you think it's fine and want to keep it as is.